libraries/AP_HAL_ChibiOS/Storage.cpp

```diff
@@ -46,8 +60,37 @@ void Storage::_storage_open(void)
     // allow for FMUv3 with no FRAM chip, fall through to flash storage
 #endif
 
+#ifdef STORAGE_FLASH_PAGE
     // load from storage backend
     _flash_load();
+#endif
+    
+#ifdef USE_POSIX
+    // allow for fallback to microSD based storage
+    sdcard_init();
+    
+    log_fd = open(HAL_STORAGE_FILE, O_RDWR|O_CREAT);
```

Comment:
Using a temporary variable here and assigning it down the bottom would remove a few lines and close off future potential bus where log_fd wasn't reset to -1 on error.

Reply:
it ends up no shorter, as log_fd also is needed as a check for having already tried to open